Tools/Scripts/webkitpy/layout_tests/controllers/single_test_runner.py

```diff
@@ -147,6 +149,23 @@ def _run_compare_test(self):
         test_result_writer.write_test_result(self._filesystem, self._port, self._results_directory, self._test_name, driver_output, expected_driver_output, test_result.failures)
         return test_result
 
+    def _run_wpt_crash_test(self):
+        driver_output = self._driver.run_test(self._driver_input(), self._stop_when_done)
+        if self._options.ignore_metrics:
+            driver_output.strip_metrics()
+
+        patterns = self._port.logging_patterns_to_strip()
+        driver_output.strip_patterns(patterns)
+        driver_output.strip_text_start_if_needed(self._port.logging_detectors_to_strip_text_start(self._driver_input().test_name))
+        driver_output.strip_stderror_patterns(self._port.stderr_patterns_to_strip())
+
+        failures = []
+        failures.extend(self._handle_error(driver_output))
+        assert(isinstance(failure, test_failures.FailureCrash) or isinstance(failure, test_failures.FailureTimeout) for failure in failures)
```

Comment:
We might need a specific handling in WTR to handle async tests like LayoutTests/imported/w3c/web-platform-tests/html/semantics/interactive-elements/the-dialog-element/showmodal-shadow-sibling-frame-crash.html.
We need to wait for class=test-wait to be removed.
This could be done as a follow-up though.

Reply:
That most certainly needs to be done in a separate follow up.

Reply:
Doing that in https://github.com/WebKit/WebKit/pull/2405